internal/generator/vector/conf_test.go

```diff
@@ -0,0 +1,228 @@
+		if diff != "" {
+			b, _ := json.MarshalIndent(e, "", " ")
+			fmt.Printf("elements:\n%s\n", string(b))
+			fmt.Println(conf)
+			fmt.Printf("diff: %s", diff)
+		}
+		Expect(diff).To(Equal(""))
+	}
+	DescribeTable("Generate full vector.toml", f,
+		Entry("with complex spec", generator.ConfGenerateTest{
+			CLSpec: logging.ClusterLoggingSpec{
+				Forwarder: &logging.ForwarderSpec{
+					Fluentd: &logging.FluentdForwarderSpec{
+						Buffer: &logging.FluentdBufferSpec{
+							ChunkLimitSize: "8m",
+							TotalLimitSize: "800000000",
+							OverflowAction: "throw_exception",
+						},
+					},
+				},
+			},
+			CLFSpec: logging.ClusterLogForwarderSpec{
+				Pipelines: []logging.PipelineSpec{
+					{
+						InputRefs: []string{
+							logging.InputNameApplication,
+							logging.InputNameInfrastructure,
+							logging.InputNameAudit},
+						OutputRefs: []string{"kafka-receiver"},
+						Name:       "pipeline",
+					},
+				},
+				Outputs: []logging.OutputSpec{
+					{
+						Type: logging.OutputTypeKafka,
+						Name: "kafka-receiver",
+						URL:  "tls://broker1-kafka.svc.messaging.cluster.local:9092/topic",
+						Secret: &logging.OutputSecretSpec{
+							Name: "kafka-receiver-1",
+						},
+					},
+				},
+			},
+			Secrets: map[string]*corev1.Secret{
+				"kafka-receiver": {
+					Data: map[string][]byte{
+						"tls.key":       []byte("junk"),
+						"tls.crt":       []byte("junk"),
+						"ca-bundle.crt": []byte("junk"),
+					},
+				},
+			},
+			ExpectedConf: `
+# Logs from containers (including openshift containers)
+[sources.raw_container_logs]
+type = "kubernetes_logs"
+auto_partial_merge = true
+exclude_paths_glob_patterns = ["/var/log/pods/openshift-logging_collector-*/*/*.log", "/var/log/pods/openshift-logging_elasticsearch-*/*/*.log", "/var/log/pods/openshift-logging_kibana-*/*/*.log"]
+
+[sources.raw_journal_logs]
+type = "journald"
+
+# Logs from host audit
+[sources.host_audit_logs]
+type = "file"
+ignore_older_secs = 600
+include = ["/var/log/audit/audit.log"]
+
+# Logs from kubernetes audit
+[sources.k8s_audit_logs]
+type = "file"
+ignore_older_secs = 600
+include = ["/var/log/kube-apiserver/audit.log"]
+
+# Logs from openshift audit
+[sources.openshift_audit_logs]
+type = "file"
+ignore_older_secs = 600
+include = ["/var/log/oauth-apiserver.audit.log"]
+
+[transforms.container_logs]
+type = "remap"
+inputs = ["raw_container_logs"]
+source = '''
+  level = "unknown"
+  if match(.message,r'(Warning|WARN|W[0-9]+|level=warn|Value:warn|"level":"warn")'){
+	level = "warn"
+  } else if match(.message, r'Info|INFO|I[0-9]+|level=info|Value:info|"level":"info"'){
+	level = "info"
+  } else if match(.message, r'Error|ERROR|E[0-9]+|level=error|Value:error|"level":"error"'){
+	level = "error"
+  } else if match(.message, r'Debug|DEBUG|D[0-9]+|level=debug|Value:debug|"level":"debug"'){
+	level = "debug"
+  }
+  .level = level
+
+  .pipeline_metadata.collector.name = "vector"
+  .pipeline_metadata.collector.version = "0.14.1"
```

Comment:
In future we probably should grab this from an ENV  since it is likely to change